blog/02-18-2023-Build-an-NFT-Full-Stack-Monster-War-Game-on-The-Celo-Blockchain/Build-an-NFT-Full-Stack-Monster-War-Game-on-The-Celo-Blockchain.md

```diff
@@ -0,0 +1,998 @@
+    });
+  } catch (error) {
+    console.log(error);
+  }
+};
+
+export const remove = async (minterContract, performActions, index) => {
+  try {
+    await performActions(async (kit) => {
+      try {
+        const { defaultAccount } = kit;
+        await minterContract.methods
+          .remove(index)
+          .send({ from: defaultAccount });
+      } catch (error) {
+        console.log({ error });
+      }
+    });
+  } catch (error) {
+    console.log(error);
+  }
+};
+
+```
+
+Now lets break down the components
+
+```javascript
+import { ethers } from "ethers";
+
+// mint an NFT
+export const createNft = async (minterContract, performActions, name) => {
+  await performActions(async (kit) => {
+    if (!name) return;
+    const { defaultAccount } = kit;
+
+    try {
+      let transaction = await minterContract.methods
+        .mint(name)
+        .send({ from: defaultAccount });
+
+      return transaction;
+    } catch (error) {
+      console.log("Error uploading file: ", error);
+    }
+  });
+};
+```
+
+First, we made the necessary import and then we declared the `createNft()` function. This function is used to mint an NFT. It takes the `minterContract` and `performActions` as arguments, along with the name of the NFT. It then uses the `defaultAccount` from the kit to mint the NFT. 
+
+```javascript
+export const getNfts = async (minterContract) => {
+  try {
+    const nfts = [];
+    const nftsLength = await minterContract.methods.getNFTlength().call();
+    // contract starts minting from index 1
+    for (let i = 0; i < Number(nftsLength); i++) {
+      const nft = new Promise(async (resolve) => {
+        const _nft = await minterContract.methods.getAllNFTS(i).call();
+        const owner = await fetchNftOwner(minterContract, i);
+        resolve({
+          index: i,
+          powerValue: _nft.powerValue,
+          name: _nft.name,
+          owner: owner,
+        });
+      });
+      nfts.push(nft);
+    }
+    return Promise.all(nfts);
+  } catch (e) {
+    console.log({ e });
+  }
+};
+```
+
+The `getNfts()` function is used to fetch all NFTs stored on the smart contract. It takes the `minterContract` as an argument and returns an array of NFTs. 
+
+```javascript
+export const fetchNftOwner = async (minterContract, index) => {
+  try {
+    return await minterContract.methods.ownerOf(index).call();
+  } catch (e) {
+    console.log({ e });
+  }
+};
+```
+ The `fetchNftOwner()` function is used to get the owner address of an NFT. It takes the `minterContract` and the index of the NFT as arguments and returns the owner address. 
+
+ ```javascript
+export const minted = async (minterContract, _address) => {
+  try {
+    return await minterContract.methods.hasMinted(_address).call();
+  } catch (e) {
+    console.log({ e });
+  }
+};
+ ```
+
+ The `minted()` function is used to check if a particular address has minted an NFT. It takes the `minterContract` and the `address` as arguments and returns a boolean. 
```

Comment:
The `minted()` function checks if a particular address has minted an NFT. It takes the `minterContract` and the `address` as arguments and returns a boolean.